lib/cfg.c

```diff
@@ -226,6 +212,20 @@ cfg_load_forced_modules(GlobalConfig *self)
     }
 }
 
+void
+cfg_discover_candidate_modules(GlobalConfig *self)
+{
+  gboolean autoload_enabled = atoi(cfg_args_get(self->globals, "autoload-compiled-modules") ? : "1");
+
+  if (self->use_plugin_discovery &&
+      autoload_enabled)
+    {
+      _sync_plugin_module_path_with_global_define(self);
+      plugin_discover_candidate_modules(&self->plugin_context);
+    }
+  cfg_load_forced_modules(self);
```

Comment:
I would not expect a discover candidate modules to load certain modules, while pushing force module load into - cfg - internal business I could relate, not in this way. Would it be okay for you to drop the commit doing it?

Reply:
Honestly I would rather get rid off the entire "forced" modules concept. But I know the reason of its existence. And if it does exist already, it should not make it into an API that needs to be called by innocent layers above.

I agree that plugin_discovery might not be the best place to call it from, but I have added a new place where discovery would run and I was under the impression that forced modules needed to be loaded in that case too. But checking the code this does not seem to be the case, so yeah, I can drop that patch.

Reply:
The revision I've just force-pushed gets rid off these patches.